firebase-perf/src/main/java/com/google/firebase/perf/metrics/HttpMetric.java

```diff
@@ -181,24 +181,21 @@ public void putAttribute(@NonNull String attribute, @NonNull String value) {
     }
   }
 
-  private void checkAttribute(@Nullable String attribute, @Nullable String value) {
+  private void checkAttribute(@Nullable String key, @Nullable String value) {
     if (isStopped) {
       throw new IllegalArgumentException(
           "HttpMetric has been logged already so unable to modify attributes");
     }
-    if (attribute == null || value == null) {
-      throw new IllegalArgumentException("Attribute must not have null key or value.");
-    }
```

Comment:
Why did you get rid of that check? The params are @Nullable but we don't verify they are non-null. Also, why are the params marked as @Nullable if we want them not to be null?

Reply:
+1 Since nulls are not valid, shouldn't it be `@NonNull`

Reply:
This is a redundant check. This is already validated in the `validateAttribute` function and so I chose to remove it to avoid redundancy. Modified to make the param "NonNull". The public facing API is already calling it as "NonNull", this was an internal check since we could trim the attribute to avoid the whitespaces in the front and back.